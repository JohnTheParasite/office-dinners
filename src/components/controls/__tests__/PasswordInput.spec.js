import { createLocalVue, shallowMount } from "@vue/test-utils"
import passwordInput from "@/components/controls/PasswordInput"
import i18n from "@/i18n"
import InputTypes from "@/enums/inputTypes"
import IconNames from "@/enums/iconNames"

describe("PasswordInput", () => {
  const localVue = new createLocalVue()
  localVue.use(i18n)

  it("PasswordInput should render properly", () => {
    let passwordElement = shallowMount(passwordInput, { localVue, i18n })
    expect(passwordElement).toMatchSnapshot()
  })

  it("toggleElementType() should work", () => {
    let passwordElement = shallowMount(passwordInput, { localVue, i18n })
    expect(passwordElement.vm.passwordInput.type).toBe(InputTypes.PASSWORD)
    expect(passwordElement.vm.passwordInput.icon).toBe(IconNames.EYE)

    passwordElement.vm.toggleElementType()
    expect(passwordElement.vm.passwordInput.type).toBe(InputTypes.TEXT)
    expect(passwordElement.vm.passwordInput.icon).toBe(IconNames.EYE_SLASH)

    passwordElement.vm.toggleElementType()
    expect(passwordElement.vm.passwordInput.type).toBe(InputTypes.PASSWORD)
    expect(passwordElement.vm.passwordInput.icon).toBe(IconNames.EYE)
  })

  it("emitInput() should work", () => {
    let passwordElement = shallowMount(passwordInput, { localVue, i18n })
    let mockfn = jest.fn()
    passwordElement.vm.$on("input", mockfn)
    passwordElement.vm.emitInput(1)

    expect(mockfn).toBeCalledWith(1, expect.anything())
  })
})
