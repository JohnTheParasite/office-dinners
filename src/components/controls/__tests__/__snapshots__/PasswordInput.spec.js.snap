// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PasswordInput PasswordInput should render properly 1`] = `<text-input-stub label="password" description="" name="" type="password" placeholder="" warningmessage="errors.fieldIsRequired" icon="fa-eye" oniconclick="[Function]"></text-input-stub>`;
